### DISCUSSION TITLE
Is there a way to increase the worker node instance type once the cluster is up and running?

### DISCUSSION THREAD

**Discussion by Aj7ay7** [kubermatic/kubeone#3478] [Q&A]
I am currently running an Kubeone cluster with master and worker nodes, where the worker nodes are using t3.medium instance types. I would like to upgrade these worker nodes to t3.large instance types without causing any disruption. Is this possible, and if so, what is the procedure to do this?

**Comment by kron4eg** [comment]
So the way to orchestrate this is to create a clone of the existing MachineDeployment you already have, but with different instanceType. Wait until new nodes will join and then cordon all old nodes (this is needed to avoid moving workloads from 1 old Node to another), and delete the old MachineDeployment which will drain old Nodes of the workloads running there, and gently move the workload to the new Nodes. 

P.S.
It will greatly help to have corresponding [PodDisruptionBudget](https://kubernetes.io/docs/tasks/run-application/configure-pdb/) objects on your workloads to help maintain some stability.
